h/security/policy/_api.py

```diff
@@ -18,10 +19,16 @@ def forget(self, *_args, **_kwargs):
     def identity(self, request) -> Identity | None:
         return self._identity_cache.get_or_create(request)
 
+    def authenticated_userid(self, request):
+        return Identity.authenticated_userid(self.identity(request))
+
     def remember(self, *_args, **_kwargs):
         # remember() isn't supported for stateless API requests.
         return []
 
+    def permits(self, request, context, permission) -> Allowed | Denied:
+        return identity_permits(self.identity(request), context, permission)
```

Comment:
Each sub-policy class that previously inherited `authenticated_userid()` and `permits()` from `IdentityBasedPolicy` now instead needs to provide these two one-liner methods to delegate `authenticated_userid()` and `permits()` to the new helpers.

As is usual with composition-over-inheritance this leads to more duplication, but this is trivial boiler-plate code and the advantage is better code readability and decoupling.